packages/shared/src/components/checklist/ChecklistStep.tsx

```diff
@@ -2,7 +2,36 @@ import React, { ReactElement } from 'react';
 import classNames from 'classnames';
 import { IconSize } from '../Icon';
 import { ArrowIcon, ChecklistAIcon } from '../icons';
-import { ChecklistStepProps } from '../../lib/checklist';
+import {
+  ChecklistCardVariant,
+  ChecklistStepProps,
+  ChecklistVariantClassNameMap,
+} from '../../lib/checklist';
+
+const iconToClassNameMap: ChecklistVariantClassNameMap = {
```

Comment:
So many differences between the versions, but I do like the solution given! 🚀

Reply:
Yeah, I don't like it but could not find better solution, good example where I think design pushes "minor" changes and then that ends up complicating stuff quite a bit on our side. We could (and even might in the future) just use the regular version.

Reply:
Yeah good points to raise, I'm not always sure if they are aware of the implication.
Maybe they'd also just say oh in that case just use what you have.

Reply:
Yeah I'm not sure on those cases as well... maybe if design really wants these and has some kind of mapping for this that an icon of this size always in mobile should be of this other size, and so for text size, would be good if we do this kind of mapping in more deeper level where we define our design system

Reply:
Thats a great idea yes, basically define `typo-sometext` something like:
- 12px on mobile
- 14px on tablet
- 16px on laptop

And then when we use `typo-sometext` it will auto resize based on the size of the screen.

Reply:
Yup, let's check with design for this. Let's not block the PR because of it though, we can fix later.